tests/test_appclass.py

```diff
@@ -39,3 +43,44 @@ def test_dependencies(generate_app):
     with pytest.raises(traitlets.TraitError):
         app.version_to_install = "v22.11.0"
     app.version_to_install = "v22.11.1"
+
+
+def test_app_watch(tmp_path):
+    """Test the aiidalab app watch responsive to the app path changes."""
+
+    @dataclass
+    class DummyApp:
+        path: Path
+        x: int = 0
+        y: int = 0
+
+        def refresh_async(self):
+            self.x += 1
+
+        def refresh(self):
+            print("refresh")
+            self.y += 1
+
+    app = DummyApp(path=Path(tmp_path))
+    app_watch = AiidaLabAppWatch(app)
+    app_watch.start()
+
+    # The observer is start in a thread so need to wait until it is alive
+    while app_watch._observer is None or not app_watch._observer.is_alive():
+        sleep(0.1)
```

Comment:
This is the trick, I didn't do this so I cannot get the persistent result previously. The observer is launched in a thread, if it is not started, the file change will not trigger the action.